include/envoy/http/codec.h

```diff
@@ -270,7 +270,9 @@ enum class StreamResetReason {
   // The stream was reset because of a resource overflow.
   Overflow,
   // Either there was an early TCP error for a CONNECT request or the peer reset with CONNECT_ERROR
-  ConnectError
+  ConnectError,
+  // Received payload did not conform to HTTP protocol.
+  ProtocolError
```

Comment:
I think this makes sense to communicate the error type, but I'm concerned it may cause problems in other places.  For example I think with this as-is we're changing some code points which depend on reset reasons without using switches such as https://github.com/envoyproxy/envoy/blob/main/source/common/http/http2/conn_pool.cc#L46

Maybe we can update that a helper switch to catch this and future additions and grep around for any othre uses of StreamResetReason which might be affected? Also let's put the changed return code in the runtime guard so if there's any cases we miss we can revert with the runtime guard.

Reply:
@alyssawilk Thanks for the comment.

What do you think about converting all multiple _if else if else ...._ to _switch ...case_? Compiler would issue warning for unhandled cases.

As for runtime overwrite I added one and it kicks in router when reply message is sent: https://github.com/envoyproxy/envoy/blob/e3f060df71a0ab532dff259397e04080a5584947/source/common/router/router.cc#L1080-L1086
There is also unit test to verify runtime guard.

Reply:
yeah, I think switch would be good - I'm a fan of compile caught errors.

I see the guard there, but I also think you might need to either add a second or move it to where you set
reason = StreamResetReason::LocalReset;
because as shown by the connection pool case, returning ProtocolError instead of whatever we used to return could have  caused a bug/regression